lib/charms/opensearch/v0/opensearch_plugins.py

```diff
@@ -275,16 +307,48 @@ def _on_update_status(self, event):
 logger = logging.getLogger(__name__)
 
 
+###########################################################################################
+# These are the basic types of errors:
+# * OpenSearchPluginError: base class for all plugin errors in the plugins code
+# * OpenSearchPluginConfigError: base class for all plugin errors related to config options
 class OpenSearchPluginError(OpenSearchError):
     """Exception thrown when an opensearch plugin is invalid."""
 
+    def __init__(self, msg, only_log=False):
+        super().__init__(msg)
+        self._only_log = only_log
+
+    @property
+    def only_log(self) -> bool:
+        """Returns the only_log property."""
+        return self._only_log
+
+
+class OpenSearchPluginConfigError(OpenSearchPluginError):
```

Comment:
Though theoretically nice, we do not need to go to this level of granularity. This can be solved and conveyed directly through the error messages in the raised exceptions

Reply:
Actually, I am making distinctions later down the road between the types of errors. I want to then treat: `repository-gcs` with a `OpenSearchPluginMissingConfigError` is not something I should be alarmed, for example.